core/services/chainlink/config.go

```
@@ -49,40 +51,171 @@ type Config struct {
+		} else if chainIDStr == "" {
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "ChainID", Msg: "required for all chains"})
+		}
+	}
+	nodes, exists := c["Nodes"]
+	if !exists {
+		err = multierr.Append(err, commonconfig.ErrMissing{Name: "Nodes", Msg: "expected at least one node"})
+	} else {
+		nodeMaps, ok := nodes.([]any)
+		if !ok {
+			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "Nodes", Value: nodes, Msg: "expected array of node configs"})
+		} else if len(nodeMaps) == 0 {
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "Nodes", Msg: "expected at least one node"})
+		} else {
+			for i, node := range nodeMaps {
+				nodeConfig, ok := node.(map[string]any)
+				if !ok {
+					err = multierr.Append(err, commonconfig.ErrInvalid{Name: fmt.Sprintf("Nodes.%d", i), Value: nodeConfig, Msg: "expected node config map"})
+				} else {
+					nodeName, exists := nodeConfig["Name"]
+					if !exists {
+						err = multierr.Append(err, commonconfig.ErrMissing{Name: fmt.Sprintf("Nodes.%d.Name", i), Msg: "required for all nodes"})
+					} else {
+						nodeNameStr, ok := nodeName.(string)
+						if !ok {
+							err = multierr.Append(err, commonconfig.ErrInvalid{Name: fmt.Sprintf("Nodes.%d.Name", i), Value: nodeName, Msg: "expected string"})
+						} else if nodeNameStr == "" {
+							err = multierr.Append(err, commonconfig.ErrEmpty{Name: fmt.Sprintf("Nodes.%d.Name", i), Msg: "required for all nodes"})
+						}
+					}
+				}
+			}
 		}
 	}
 	return err
 }
 
-func (c *RawConfig) IsEnabled() bool {
-	if c == nil {
-		return false
-	}
-
-	enabled, ok := (*c)["Enabled"].(bool)
+func (c RawConfig) IsEnabled() bool {
+	enabled, ok := c["Enabled"].(bool)
 	return ok && enabled
 }
 
-func (c *RawConfig) ChainID() string {
-	if c == nil {
-		return ""
+func (c RawConfig) ChainID() string {
+	chainID, _ := c["ChainID"].(string)
+	return chainID
+}
+
+func (c *RawConfig) SetFrom(config RawConfig) error {
```

Why not check errors from this func which returns error? This seem unnecessarily brittle.

---

can you specify which errors you're talking about? if it's about L192-193, the errors would already have been checked in ValidateConfig.

---

`ValidateConfig` is called on the end result, after `SetFrom` is applied to the raw input.

---

ValidateConfig is called on the new config on L169. any config that was appended to the RawConfig would already have been validated since it would have been added in SetFrom.
this is the same pattern in other SetFrom functions. can you explain why this is brittle?

---

ValidateConfig is already called from elsewhere, on the end result. It cannot be used on partial configs which may be incomplete/invalid on their own.

It is always brittle to ignore errors. If you truly have confidence that they will "never" happen, then you should at least `panic`. But if you already parsed them elsewhere and checked the error at the time, then you should retain the parsed value to use instead of re-parsing.